File: iris-ld/iris-did/src/main/java/city/smartb/iris/did/model/DIDAuthentication.java
package city.smartb.iris.did.model;

public class DIDAuthentication {

	private final Object json;

	public DIDAuthentication(Object json) {
		this.json = json;
	}

	public String getPublicKeyId() {
		if(json instanceof String) {
			return (String) json;
		}
		if(json instanceof DIDVerificationMethod) {
			return ((DIDVerificationMethod) json).getId();
		}
		return null;
	}

	public Object toJSON() {
		return json;
	}

}

File: iris-ld/iris-did/src/main/kotlin/city/smartb/iris/ld/did/DIDAuthentication.kt
package city.smartb.iris.ld.did

class DIDAuthentication(private val json: Any) {
    val publicKeyId: String?
        get() {
            if (json is String) {
                return json
            }
            return if (json is DIDVerificationMethod) {
                json.id
            } else null
        }

    fun toJSON(): Any {
        return json
    }
}
